test-manager/src/tests/mod.rs

```diff
@@ -115,1259 +119,1318 @@ macro_rules! get_possible_api_endpoints {
+    // install package
+    let mut ctx = context::current();
+    ctx.deadline = SystemTime::now().checked_add(INSTALL_TIMEOUT).unwrap();
+
+    rpc.install_app(ctx, get_package_desc(&rpc, &*CURRENT_APP_FILENAME).await?)
+        .await?
+        .map_err(|err| Error::Package("current app", err))?;
+
+    // verify that daemon is running
+    if rpc.mullvad_daemon_get_status(context::current()).await? != ServiceStatus::Running {
+        return Err(Error::DaemonNotRunning);
     }
 
-    /// Verify that outgoing TCP, UDP, and ICMP packets can be observed
-    /// in the disconnected state. The purpose is mostly to rule prevent
-    /// false negatives in other tests.
-    #[manager_test(priority = -101)]
-    pub async fn test_disconnected_state(
-        rpc: ServiceClient,
-        mut mullvad_client: ManagementServiceClient,
-    ) -> Result<(), Error> {
-        let inet_destination = "1.3.3.7:1337".parse().unwrap();
-
-        log::info!("Verify tunnel state: disconnected");
-        assert_tunnel_state!(&mut mullvad_client, TunnelState::Disconnected);
-
-        //
-        // Test whether outgoing packets can be observed
-        //
-
-        log::info!("Sending packets to {inet_destination}");
-
-        let detected_probes =
-            send_guest_probes(rpc.clone(), Some(Interface::NonTunnel), inet_destination).await?;
-        assert!(
-            detected_probes.all(),
-            "did not see (all) outgoing packets to destination: {detected_probes:?}",
-        );
+    Ok(())
+}
 
-        Ok(())
+async fn get_package_desc(rpc: &ServiceClient, name: &str) -> Result<Package, Error> {
+    match rpc.get_os(context::current()).await.map_err(Error::Rpc)? {
+        meta::Os::Linux => Ok(Package {
+            r#type: PackageType::Dpkg,
+            path: Path::new(&format!("/opt/testing/{}", name)).to_path_buf(),
+        }),
+        meta::Os::Windows => Ok(Package {
+            r#type: PackageType::NsisExe,
+            path: Path::new(&format!(r"E:\{}", name)).to_path_buf(),
+        }),
+        _ => unimplemented!(),
     }
+}
 
-    /// Log in and create a new device
-    /// from the account.
-    #[manager_test(priority = -101)]
-    pub async fn test_login(
-        _rpc: ServiceClient,
-        mut mullvad_client: ManagementServiceClient,
-    ) -> Result<(), Error> {
-        // TODO: Test too many devices, removal, etc.
+/// This is not a test per-se but gets the default settings from the daemon after the newest
+/// version has been installed and sets these as the DEFAULT_SETTINGS. This is used by the test
+/// cleanup function after each other test is run.
+/// It is important that this test has a priority that makes it run right after the test which
+/// installs the newest version.
+/// It is also important that all tests that run before this one set `cleanup = false` in order to
+/// avoid panicking when the default have not yet been set.
+#[test_function(priority = -101)]
+pub async fn set_default_settings(
```

Comment:
Running it before the tests is a good idea I will change that. I'm not sure what you mean by assuming that settings are default. Do you mean that `cleanup_after_test` would be pulling down the settings from the daemon everytime it ran and using a `.get_or_init` to only set it the first time?
The potential issues I see with that is most significantly that someone could create a test which uses the old mullvad management interface which runs before the last installation. This would cause us to run the cleanup with the old mullvad management interface and get the old settings. 
However I could perhaps make sure the settings are not set when the old interface is used but if they are set then the cleanup function is run.

I think I'll do that. I'll create a separate function which sets the settings if they aren't already set but only does this when we are using the new management interface and then I'll run the cleanup only when the default settings are set.

Reply:
> Do you mean that cleanup_after_test would be pulling down the settings from the daemon everytime it ran and using a .get_or_init to only set it the first time?

Yes. And possibly check whether it's already set first to avoid unnecessarily fetching the settings.